app/src/main/java/com/onboarding/parkingsystemkotlin/mvp/presenter/ReservePresenter.kt

```diff
@@ -12,11 +17,45 @@ class ReservePresenter (private val view: ReserveActivityContract.ReserveView):
     }
 
     override fun onOkButtonPress() {
-        view.showReserveSavedToast() //placeholder for next feature
-        view.returnToMainActivity()
+        model.setReservationInfo(view.getParkingLot(), view.getUserPassword())
+        val reservation = model.getReservation()
+        when (model.reserveFieldsCheck(reservation)) {
+            ReserveComprobation.MISSING_START -> view.showMissingStartDateToast()
+            ReserveComprobation.MISSING_END -> view.showMissingEndDateToast()
+            ReserveComprobation.MISSING_LOT -> view.showMissingParkingLotToast()
+            ReserveComprobation.MISSING_PASSWORD -> view.showMissingUserPasswordToast()
+            ReserveComprobation.RESERVATION_OVERLAP -> view.showReservationOverlapToast()
+            ReserveComprobation.COMPROBATION_OK -> {
+                view.showReserveSavedToast()
+                model.addReservation(reservation)
+                view.returnToMainActivity()
+            }
+        }
     }
 
     override fun onCancelButtonPress() {
         view.returnToMainActivity()
     }
+
+    override fun setReservationStartDate(startDate: Calendar) {
+        model.setStartDate(startDate)
+        view.setStartDateTextView(getDateString(model.getStartDate()))
+    }
+
+    override fun setReservationEndDate(endDate: Calendar) {
+        model.setEndDate(endDate)
+        view.setEndDateTextView(getDateString(model.getEndDate()))
+    }
+
+    private fun getDateString(date: Calendar?): String {
+        var ret = ConstantUtils.EMPTY_STRING
+        date?.let {
+            ret = (it.get(Calendar.DAY_OF_MONTH).toString() + ConstantUtils.SLASH +
```

Comment:
In kotlin to concat you can do this, for example you have this: 
val a: String = "Hola"
val b: String = "Mundo"
val c: String = "$a $b"
and avoid using val c: String = a + " " + b

Reply:
done